packages/authentication/test/acceptance/basic-auth.ts

```diff
@@ -0,0 +1,206 @@
+
+const BasicStrategy = require('passport-http').BasicStrategy;
+
+describe('Basic Authentication', () => {
+  let app: Application;
+  let users: UserRepository;
+
+  beforeEach(givenUserRespository);
+  beforeEach(givenAnApplication);
+  beforeEach(givenControllerInApp);
+  beforeEach(givenAuthenticatedSequence);
+  beforeEach(givenProviders);
+
+  it ('authenticates successfully for correct credentials', () => {
+    return whenIMakeRequestTo(app).then(client => {
+      const credential =
+        users.list.joe.profile.id + ':' + users.list.joe.password;
+      const hash = new Buffer(credential).toString('base64');
+      return client.get('/whoAmI')
+        .set('Authorization', 'Basic ' + hash)
+        .expect(users.list.joe.profile.id);
+    });
+  });
+
+  it('returns error for invalid credentials', () => {
+    return whenIMakeRequestTo(app).then(client => {
+      const credential =
+        users.list.Simpson.profile.id + ':' + 'invalid';
+      const hash = new Buffer(credential).toString('base64');
+      return client.get('/whoAmI')
+        .set('Authorization', 'Basic ' + hash)
+        .expect(401);
+    });
+  });
+
+  function givenUserRespository() {
+    users = new UserRepository({
+      joe : {profile: {id: 'joe'}, password: '12345'},
+      Simpson: {profile: {id: 'sim123'}, password: 'alpha'},
+      Flintstone: {profile: {id: 'Flint'}, password: 'beta'},
+      George: {profile: {id: 'Curious'}, password: 'gamma'},
+    });
+  }
+
+  function givenAnApplication() {
+    app = new Application();
+    app.bind('application.name').to('SequenceApp');
+  }
+
+  function givenControllerInApp() {
+    const apispec = givenOpenApiSpec()
+      .withOperation('get', '/whoAmI', {
+        'x-operation-name': 'whoAmI',
+        responses: {
+          '200': {
+            type: 'string',
+          },
+        },
+      })
+      .build();
+
+    @api(apispec)
+    class MyController {
+      constructor(@inject('authentication.user') private user: UserProfile) {}
+
+      @authenticate('BasicStrategy')
+      async whoAmI() : Promise<string> {
+        return this.user.id;
+      }
+    }
+    app.controller(MyController);
+  }
+
+  function givenAuthenticatedSequence() {
+    class MySequence {
+      constructor(
+        @inject('findRoute') protected findRoute: FindRoute,
+        @inject('getFromContext') protected getFromContext: getFromContext,
+        @inject('invokeMethod') protected invoke: InvokeMethod,
+        @inject('bindElement') protected bindElement: bindElement,
+      ) {}
+
+      async run(req: ParsedRequest, res: ServerResponse) {
+        try {
+          const {
+            controller,
+            methodName,
+            spec: routeSpec,
+            pathParams,
+          } = this.findRoute(req);
+
+          // Resolve authenticate() from AuthenticationProvider
+          const authenticate: AuthenticateFn =
+            await this.getFromContext(BindingKeys.Authentication.PROVIDER);
```

Comment:
This needs to be refactored. I think the missing piece is `OperationMeta` which should be the result of another `Action`. This line would then become:

```
const meta = this.getOperationMeta();
const user: UserProfile = await authenticate(request, meta); // injected
```